integration/cmd/historicalspeed-timezone-builder/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"flag"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/qedus/osmpbf"
+
+	"github.com/golang/glog"
+)
+
+func main() {
+	flag.Parse()
+
+	if flags.pbfSource != pbfSourceUniDB {
+		glog.Errorf("Only support %s at the moment.", pbfSourceUniDB)
+		os.Exit(1)
+		return
+	}
+
+	startTime := time.Now()
+
+	osmWaysChan := make(chan *osmpbf.Way)
+	wayTimezoneInfoChan := make(chan *wayTimezoneInfo)
+	errChan := make(chan error)
+
+	go func() {
+		glog.V(1).Infof("ways mapping updater routine start.")
+		defer glog.V(1).Info("ways mapping updater routine exited.")
+
+		errChan <- newWaysMappingUpdater(strings.Split(flags.inWaysMappingFile, ","), flags.outWaysMappingFile, flags.withCSVHeader, wayTimezoneInfoChan)
+	}()
+	go func() {
+		glog.V(1).Infof("pbf parser routine start.")
+		defer glog.V(1).Info("pbf parser routine exited.")
+
+		errChan <- newPBFParser(flags.pbf, osmWaysChan)
+		close(osmWaysChan)
+	}()
+	go func() {
+		glog.V(1).Infof("timezone builder for unidb routine start.")
+		defer glog.V(1).Info("timezone builder for unidb routine exited.")
+
+		newTimezoneBuilderForUniDB(osmWaysChan, wayTimezoneInfoChan)
+		close(wayTimezoneInfoChan)
+		errChan <- nil
+	}()
+
+	for i := 0; i < 3; i++ {
```

Comment:
nit: I think using `select` could achieve same target, but could avoid `3` in case the number of `goroutine` has been adjusted.

Reply:
It can not. `select` only can pick up active one from multiple channels, but it don't know how many to wait. The usage is something like `wg.WaitGroup()`, but also handles error.

Reply:
How about following logic:
```go
select {
    case err1  <- chan1:
               // error handling
    case err2 <- chan2:
               // error handling
    case <- time.After(timeout_value * time.Second):
               // xxxx
}
```

Reply:
There'll need `3` different `chan error` in this logic, isn't it?